src/realm/sync/subscriptions.cpp

```diff
@@ -973,4 +971,25 @@ int64_t SubscriptionStore::set_active_as_latest(Transaction& wt)
     return version;
 }
 
+int64_t SubscriptionStore::mark_active_as_complete(Transaction& wt)
```

Comment:
you could make this a no-op if the state is already Complete (unless it's possible to set the state and not deliver the notifications)

Reply:
I think if the active set is already complete then `m_pending_notifications` should be empty barring someone waiting for a Superseded notification, so the only unnecessary thing we're doing in that scenario is acquiring the lock.